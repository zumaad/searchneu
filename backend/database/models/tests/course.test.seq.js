import { sequelize, Course, Section } from '../index';
import elastic from '../../../elastic';

beforeEach(async () => {
  await Section.truncate({ cascade: true, restartIdentity: true });
  await Course.truncate({ cascade: true, restartIdentity: true });


  jest.spyOn(elastic, 'bulkIndexFromMap').mockImplementation(() => {});

  await Course.create({
    id: 'neu.edu/202030/CS/2500',
    host: 'neu.edu',
    classId: '2500',
    name: 'Fundamentals of Computer Science 1',
    termId: '202030',
    subject: 'CS',
    maxCredits: 4,
    minCredits: 4,
    desc: 'a good class',
    url: 'https://foo.com',
    prettyUrl: 'https://foo.com',
    lastUpdateTime: 123456789,
    classAttributes: ['hebloo'],
  });

  await Section.create({
    id: 'neu.edu/202030/CS/2500/19350',
    classHash: 'neu.edu/202030/CS/2500',
    seatsCapacity: 80,
    seatsRemaining: 0,
    waitCapacity: 0,
    waitRemaining: 0,
    online: false,
    honors: false,
    url: 'https://foo.com/19350',
    crn: '19350',
    meetings: [
      {
        startDate: 17903,
        endDate: 18013,
        profs: [
          'Benjamin Lerner',
        ],
        where: 'IV 010',
        type: 'Class',
        times: {
          4: [
            {
              start: 33600,
              end: 41400,
            },
          ],
        },
        allProfs: [
          'Benjamin Lerner',
          'Alan Mislove',
        ],
      },
    ],
  });

  await Section.create({
    id: 'neu.edu/202030/CS/2500/19360',
    classHash: 'neu.edu/202030/CS/2500',
    seatsCapacity: 80,
    seatsRemaining: 5,
    waitCapacity: 10,
    waitRemaining: 2,
    online: false,
    honors: false,
    url: 'https://foo.com/19360',
    crn: '19360',
    meetings: [
      {
        startDate: 17903,
        endDate: 18013,
        profs: [
          'Alan Mislove',
        ],
        where: 'West Village G 010',
        type: 'Class',
        times: {
          2: [
            {
              start: 33600,
              end: 41400,
            },
          ],
        },
        allProfs: [
          'Benjamin Lerner',
          'Alan Mislove',
        ],
      },
    ],
  });
});

afterAll(async () => {
  await sequelize.close();
});

describe('bulkJSON', () => {
  it('generates proper JSON for a class', async () => {
    const course = await Course.findByPk('neu.edu/202030/CS/2500');
    const results = await Course.bulkJSON([course]);
    expect(results).toMatchSnapshot();
  });

  it('includes prerequisites and corequisites if they exist', async () => {
    // obviously...
    const prereqs = {
      type: 'and',
      values: [
        {
          subject: 'CS',
          classId: '2510',
        },
      ],
    };

    const coreqs = {
      type: 'and',
      values: [
        {
          subject: 'CS',
          classId: '2501',
        },
      ],
    };

    const prereqsFor = {
      values: [
        {
          subject: 'CS',
          classId: '2510',
        },
        {
          subject: 'CS',
          classId: '2800',
        },
      ],
    };

    const optPrereqsFor = {
      values: [
        {
          subject: 'CS',
          classId: '3200',
        },
      ],
    };

    await Course.update({
      prereqs: prereqs,
      coreqs: coreqs,
      prereqsFor: prereqsFor,
      optPrereqsFor: optPrereqsFor,
    }, { where: { id: 'neu.edu/202030/CS/2500' }, hooks: false });

    const course = await Course.findByPk('neu.edu/202030/CS/2500');
    const results = await Course.bulkJSON([course]);

    expect(results).toMatchSnapshot();
  });
});

describe('bulkUpsertES', () => {
  it('upserts to ES', async () => {
    const course = await Course.findByPk('neu.edu/202030/CS/2500');
    await Course.bulkUpsertES([course]);

    expect(elastic.bulkIndexFromMap.mock.calls[0]).toMatchSnapshot();
  });
});

describe('afterBulkCreate', () => {
  it('updates ES', async () => {
    await Course.bulkCreate([{ name: 'Fundies 1', id: 'neu.edu/202030/CS/2500' }], { where: { id: 'neu.edu/202030/CS/2500' }, updateOnDuplicate: ['name'] });
    expect(elastic.bulkIndexFromMap.mock.calls[0]).toMatchSnapshot();
  });
});

describe('afterBulkUpdate', () => {
  it('updates ES', async () => {
    await Course.update({ name: 'Fundies 1' }, { where: { id: 'neu.edu/202030/CS/2500' } });
    expect(elastic.bulkIndexFromMap.mock.calls[0]).toMatchSnapshot();
  });
});

describe('toJSON', () => {
  it('generates the correct object', async () => {
    const course = await Course.findByPk('neu.edu/202030/CS/2500');
    expect(course.toJSON()).toEqual({
      classAttributes: ['hebloo'],
      maxCredits: 4,
      minCredits: 4,
      desc: 'a good class',
      classId: '2500',
      prettyUrl: 'https://foo.com',
      url: 'https://foo.com',
      name: 'Fundamentals of Computer Science 1',
      lastUpdateTime: 123456789,
      termId: '202030',
      host: 'neu.edu',
      subject: 'CS',
    });
  });
});
