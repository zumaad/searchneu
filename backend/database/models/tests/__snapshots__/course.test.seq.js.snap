// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`afterBulkCreate updates ES 1`] = `
Array [
  "classes",
  Object {
    "neu.edu/202030/CS/2500": Object {
      "class": Object {
        "classAttributes": Array [
          "hebloo",
        ],
        "classId": "2500",
        "crns": Array [
          "19350",
          "19360",
        ],
        "desc": "a good class",
        "host": "neu.edu",
        "lastUpdateTime": 123456789,
        "maxCredits": 4,
        "minCredits": 4,
        "name": "Fundamentals of Computer Science 1",
        "prettyUrl": "https://foo.com",
        "sections": Array [
          Object {
            "classId": "2500",
            "crn": "19350",
            "honors": false,
            "host": "neu.edu",
            "lastUpdateTime": 123456789,
            "meetings": Array [
              Object {
                "allProfs": Array [
                  "Benjamin Lerner",
                  "Alan Mislove",
                ],
                "endDate": 18013,
                "profs": Array [
                  "Benjamin Lerner",
                ],
                "startDate": 17903,
                "times": Object {
                  "4": Array [
                    Object {
                      "end": 41400,
                      "start": 33600,
                    },
                  ],
                },
                "type": "Class",
                "where": "IV 010",
              },
            ],
            "online": false,
            "seatsCapacity": 80,
            "seatsRemaining": 0,
            "subject": "CS",
            "termId": "202030",
            "url": "https://foo.com/19350",
            "waitCapacity": 0,
            "waitRemaining": 0,
          },
          Object {
            "classId": "2500",
            "crn": "19360",
            "honors": false,
            "host": "neu.edu",
            "lastUpdateTime": 123456789,
            "meetings": Array [
              Object {
                "allProfs": Array [
                  "Benjamin Lerner",
                  "Alan Mislove",
                ],
                "endDate": 18013,
                "profs": Array [
                  "Alan Mislove",
                ],
                "startDate": 17903,
                "times": Object {
                  "2": Array [
                    Object {
                      "end": 41400,
                      "start": 33600,
                    },
                  ],
                },
                "type": "Class",
                "where": "West Village G 010",
              },
            ],
            "online": false,
            "seatsCapacity": 80,
            "seatsRemaining": 5,
            "subject": "CS",
            "termId": "202030",
            "url": "https://foo.com/19360",
            "waitCapacity": 10,
            "waitRemaining": 2,
          },
        ],
        "subject": "CS",
        "termId": "202030",
        "url": "https://foo.com",
      },
      "sections": Array [
        Object {
          "classId": "2500",
          "crn": "19350",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Benjamin Lerner",
              ],
              "startDate": 17903,
              "times": Object {
                "4": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "IV 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 0,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19350",
          "waitCapacity": 0,
          "waitRemaining": 0,
        },
        Object {
          "classId": "2500",
          "crn": "19360",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Alan Mislove",
              ],
              "startDate": 17903,
              "times": Object {
                "2": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "West Village G 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 5,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19360",
          "waitCapacity": 10,
          "waitRemaining": 2,
        },
      ],
      "type": "class",
    },
  },
]
`;

exports[`afterBulkUpdate updates ES 1`] = `
Array [
  "classes",
  Object {
    "neu.edu/202030/CS/2500": Object {
      "class": Object {
        "classAttributes": Array [
          "hebloo",
        ],
        "classId": "2500",
        "crns": Array [
          "19350",
          "19360",
        ],
        "desc": "a good class",
        "host": "neu.edu",
        "lastUpdateTime": 123456789,
        "maxCredits": 4,
        "minCredits": 4,
        "name": "Fundamentals of Computer Science 1",
        "prettyUrl": "https://foo.com",
        "sections": Array [
          Object {
            "classId": "2500",
            "crn": "19350",
            "honors": false,
            "host": "neu.edu",
            "lastUpdateTime": 123456789,
            "meetings": Array [
              Object {
                "allProfs": Array [
                  "Benjamin Lerner",
                  "Alan Mislove",
                ],
                "endDate": 18013,
                "profs": Array [
                  "Benjamin Lerner",
                ],
                "startDate": 17903,
                "times": Object {
                  "4": Array [
                    Object {
                      "end": 41400,
                      "start": 33600,
                    },
                  ],
                },
                "type": "Class",
                "where": "IV 010",
              },
            ],
            "online": false,
            "seatsCapacity": 80,
            "seatsRemaining": 0,
            "subject": "CS",
            "termId": "202030",
            "url": "https://foo.com/19350",
            "waitCapacity": 0,
            "waitRemaining": 0,
          },
          Object {
            "classId": "2500",
            "crn": "19360",
            "honors": false,
            "host": "neu.edu",
            "lastUpdateTime": 123456789,
            "meetings": Array [
              Object {
                "allProfs": Array [
                  "Benjamin Lerner",
                  "Alan Mislove",
                ],
                "endDate": 18013,
                "profs": Array [
                  "Alan Mislove",
                ],
                "startDate": 17903,
                "times": Object {
                  "2": Array [
                    Object {
                      "end": 41400,
                      "start": 33600,
                    },
                  ],
                },
                "type": "Class",
                "where": "West Village G 010",
              },
            ],
            "online": false,
            "seatsCapacity": 80,
            "seatsRemaining": 5,
            "subject": "CS",
            "termId": "202030",
            "url": "https://foo.com/19360",
            "waitCapacity": 10,
            "waitRemaining": 2,
          },
        ],
        "subject": "CS",
        "termId": "202030",
        "url": "https://foo.com",
      },
      "sections": Array [
        Object {
          "classId": "2500",
          "crn": "19350",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Benjamin Lerner",
              ],
              "startDate": 17903,
              "times": Object {
                "4": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "IV 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 0,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19350",
          "waitCapacity": 0,
          "waitRemaining": 0,
        },
        Object {
          "classId": "2500",
          "crn": "19360",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Alan Mislove",
              ],
              "startDate": 17903,
              "times": Object {
                "2": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "West Village G 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 5,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19360",
          "waitCapacity": 10,
          "waitRemaining": 2,
        },
      ],
      "type": "class",
    },
  },
]
`;

exports[`bulkJSON generates proper JSON for a class 1`] = `
Object {
  "neu.edu/202030/CS/2500": Object {
    "class": Object {
      "classAttributes": Array [
        "hebloo",
      ],
      "classId": "2500",
      "crns": Array [
        "19350",
        "19360",
      ],
      "desc": "a good class",
      "host": "neu.edu",
      "lastUpdateTime": 123456789,
      "maxCredits": 4,
      "minCredits": 4,
      "name": "Fundamentals of Computer Science 1",
      "prettyUrl": "https://foo.com",
      "sections": Array [
        Object {
          "classId": "2500",
          "crn": "19350",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Benjamin Lerner",
              ],
              "startDate": 17903,
              "times": Object {
                "4": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "IV 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 0,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19350",
          "waitCapacity": 0,
          "waitRemaining": 0,
        },
        Object {
          "classId": "2500",
          "crn": "19360",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Alan Mislove",
              ],
              "startDate": 17903,
              "times": Object {
                "2": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "West Village G 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 5,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19360",
          "waitCapacity": 10,
          "waitRemaining": 2,
        },
      ],
      "subject": "CS",
      "termId": "202030",
      "url": "https://foo.com",
    },
    "sections": Array [
      Object {
        "classId": "2500",
        "crn": "19350",
        "honors": false,
        "host": "neu.edu",
        "lastUpdateTime": 123456789,
        "meetings": Array [
          Object {
            "allProfs": Array [
              "Benjamin Lerner",
              "Alan Mislove",
            ],
            "endDate": 18013,
            "profs": Array [
              "Benjamin Lerner",
            ],
            "startDate": 17903,
            "times": Object {
              "4": Array [
                Object {
                  "end": 41400,
                  "start": 33600,
                },
              ],
            },
            "type": "Class",
            "where": "IV 010",
          },
        ],
        "online": false,
        "seatsCapacity": 80,
        "seatsRemaining": 0,
        "subject": "CS",
        "termId": "202030",
        "url": "https://foo.com/19350",
        "waitCapacity": 0,
        "waitRemaining": 0,
      },
      Object {
        "classId": "2500",
        "crn": "19360",
        "honors": false,
        "host": "neu.edu",
        "lastUpdateTime": 123456789,
        "meetings": Array [
          Object {
            "allProfs": Array [
              "Benjamin Lerner",
              "Alan Mislove",
            ],
            "endDate": 18013,
            "profs": Array [
              "Alan Mislove",
            ],
            "startDate": 17903,
            "times": Object {
              "2": Array [
                Object {
                  "end": 41400,
                  "start": 33600,
                },
              ],
            },
            "type": "Class",
            "where": "West Village G 010",
          },
        ],
        "online": false,
        "seatsCapacity": 80,
        "seatsRemaining": 5,
        "subject": "CS",
        "termId": "202030",
        "url": "https://foo.com/19360",
        "waitCapacity": 10,
        "waitRemaining": 2,
      },
    ],
    "type": "class",
  },
}
`;

exports[`bulkJSON includes prerequisites and corequisites if they exist 1`] = `
Object {
  "neu.edu/202030/CS/2500": Object {
    "class": Object {
      "classAttributes": Array [
        "hebloo",
      ],
      "classId": "2500",
      "coreqs": Object {
        "type": "and",
        "values": Array [
          Object {
            "classId": "2501",
            "subject": "CS",
          },
        ],
      },
      "crns": Array [
        "19350",
        "19360",
      ],
      "desc": "a good class",
      "host": "neu.edu",
      "lastUpdateTime": 123456789,
      "maxCredits": 4,
      "minCredits": 4,
      "name": "Fundamentals of Computer Science 1",
      "optPrereqsFor": Object {
        "values": Array [
          Object {
            "classId": "3200",
            "subject": "CS",
          },
        ],
      },
      "prereqs": Object {
        "type": "and",
        "values": Array [
          Object {
            "classId": "2510",
            "subject": "CS",
          },
        ],
      },
      "prereqsFor": Object {
        "values": Array [
          Object {
            "classId": "2510",
            "subject": "CS",
          },
          Object {
            "classId": "2800",
            "subject": "CS",
          },
        ],
      },
      "prettyUrl": "https://foo.com",
      "sections": Array [
        Object {
          "classId": "2500",
          "crn": "19350",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Benjamin Lerner",
              ],
              "startDate": 17903,
              "times": Object {
                "4": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "IV 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 0,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19350",
          "waitCapacity": 0,
          "waitRemaining": 0,
        },
        Object {
          "classId": "2500",
          "crn": "19360",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Alan Mislove",
              ],
              "startDate": 17903,
              "times": Object {
                "2": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "West Village G 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 5,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19360",
          "waitCapacity": 10,
          "waitRemaining": 2,
        },
      ],
      "subject": "CS",
      "termId": "202030",
      "url": "https://foo.com",
    },
    "sections": Array [
      Object {
        "classId": "2500",
        "crn": "19350",
        "honors": false,
        "host": "neu.edu",
        "lastUpdateTime": 123456789,
        "meetings": Array [
          Object {
            "allProfs": Array [
              "Benjamin Lerner",
              "Alan Mislove",
            ],
            "endDate": 18013,
            "profs": Array [
              "Benjamin Lerner",
            ],
            "startDate": 17903,
            "times": Object {
              "4": Array [
                Object {
                  "end": 41400,
                  "start": 33600,
                },
              ],
            },
            "type": "Class",
            "where": "IV 010",
          },
        ],
        "online": false,
        "seatsCapacity": 80,
        "seatsRemaining": 0,
        "subject": "CS",
        "termId": "202030",
        "url": "https://foo.com/19350",
        "waitCapacity": 0,
        "waitRemaining": 0,
      },
      Object {
        "classId": "2500",
        "crn": "19360",
        "honors": false,
        "host": "neu.edu",
        "lastUpdateTime": 123456789,
        "meetings": Array [
          Object {
            "allProfs": Array [
              "Benjamin Lerner",
              "Alan Mislove",
            ],
            "endDate": 18013,
            "profs": Array [
              "Alan Mislove",
            ],
            "startDate": 17903,
            "times": Object {
              "2": Array [
                Object {
                  "end": 41400,
                  "start": 33600,
                },
              ],
            },
            "type": "Class",
            "where": "West Village G 010",
          },
        ],
        "online": false,
        "seatsCapacity": 80,
        "seatsRemaining": 5,
        "subject": "CS",
        "termId": "202030",
        "url": "https://foo.com/19360",
        "waitCapacity": 10,
        "waitRemaining": 2,
      },
    ],
    "type": "class",
  },
}
`;

exports[`bulkUpsertES upserts to ES 1`] = `
Array [
  "classes",
  Object {
    "neu.edu/202030/CS/2500": Object {
      "class": Object {
        "classAttributes": Array [
          "hebloo",
        ],
        "classId": "2500",
        "crns": Array [
          "19350",
          "19360",
        ],
        "desc": "a good class",
        "host": "neu.edu",
        "lastUpdateTime": 123456789,
        "maxCredits": 4,
        "minCredits": 4,
        "name": "Fundamentals of Computer Science 1",
        "prettyUrl": "https://foo.com",
        "sections": Array [
          Object {
            "classId": "2500",
            "crn": "19350",
            "honors": false,
            "host": "neu.edu",
            "lastUpdateTime": 123456789,
            "meetings": Array [
              Object {
                "allProfs": Array [
                  "Benjamin Lerner",
                  "Alan Mislove",
                ],
                "endDate": 18013,
                "profs": Array [
                  "Benjamin Lerner",
                ],
                "startDate": 17903,
                "times": Object {
                  "4": Array [
                    Object {
                      "end": 41400,
                      "start": 33600,
                    },
                  ],
                },
                "type": "Class",
                "where": "IV 010",
              },
            ],
            "online": false,
            "seatsCapacity": 80,
            "seatsRemaining": 0,
            "subject": "CS",
            "termId": "202030",
            "url": "https://foo.com/19350",
            "waitCapacity": 0,
            "waitRemaining": 0,
          },
          Object {
            "classId": "2500",
            "crn": "19360",
            "honors": false,
            "host": "neu.edu",
            "lastUpdateTime": 123456789,
            "meetings": Array [
              Object {
                "allProfs": Array [
                  "Benjamin Lerner",
                  "Alan Mislove",
                ],
                "endDate": 18013,
                "profs": Array [
                  "Alan Mislove",
                ],
                "startDate": 17903,
                "times": Object {
                  "2": Array [
                    Object {
                      "end": 41400,
                      "start": 33600,
                    },
                  ],
                },
                "type": "Class",
                "where": "West Village G 010",
              },
            ],
            "online": false,
            "seatsCapacity": 80,
            "seatsRemaining": 5,
            "subject": "CS",
            "termId": "202030",
            "url": "https://foo.com/19360",
            "waitCapacity": 10,
            "waitRemaining": 2,
          },
        ],
        "subject": "CS",
        "termId": "202030",
        "url": "https://foo.com",
      },
      "sections": Array [
        Object {
          "classId": "2500",
          "crn": "19350",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Benjamin Lerner",
              ],
              "startDate": 17903,
              "times": Object {
                "4": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "IV 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 0,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19350",
          "waitCapacity": 0,
          "waitRemaining": 0,
        },
        Object {
          "classId": "2500",
          "crn": "19360",
          "honors": false,
          "host": "neu.edu",
          "lastUpdateTime": 123456789,
          "meetings": Array [
            Object {
              "allProfs": Array [
                "Benjamin Lerner",
                "Alan Mislove",
              ],
              "endDate": 18013,
              "profs": Array [
                "Alan Mislove",
              ],
              "startDate": 17903,
              "times": Object {
                "2": Array [
                  Object {
                    "end": 41400,
                    "start": 33600,
                  },
                ],
              },
              "type": "Class",
              "where": "West Village G 010",
            },
          ],
          "online": false,
          "seatsCapacity": 80,
          "seatsRemaining": 5,
          "subject": "CS",
          "termId": "202030",
          "url": "https://foo.com/19360",
          "waitCapacity": 10,
          "waitRemaining": 2,
        },
      ],
      "type": "class",
    },
  },
]
`;
